runtime/lua/vim/treesitter.lua

```diff
@@ -127,7 +127,7 @@ function M.get_parser(bufnr, lang, opts)
       )
     end
   elseif parsers[bufnr] == nil or parsers[bufnr]:lang() ~= lang then
-    parsers[bufnr] = M._create_parser(bufnr, lang, opts)
+    parsers[bufnr] = M._create_parser(bufnr, lang --[[@as string]], opts)
```

Comment:
This shouldn't be needed. Lang must be losing its type further up.

Reply:
It's not that it's losing it's type, but the parameter type is `string|nil` but the checks above should ensure that in here we have a valid `lang`.

Reply:
If you need to remove `nil` from a type then best to use `assert(var)`.